src/Npgsql/BackendMessages/RowDescriptionMessage.cs

```diff
@@ -360,33 +362,54 @@ internal void GetInfo(Type? type, ref ColumnInfo lastColumnInfo)
         [MethodImpl(MethodImplOptions.NoInlining)]
         void GetInfoSlow(Type? type, out ColumnInfo lastColumnInfo)
         {
-            var typeInfo = AdoSerializerHelpers.GetTypeInfoForReading(type ?? typeof(object), PostgresType, _serializerOptions);
+            var (typeInfo, exception) = AdoSerializerHelpers.TryGetTypeInfoForReading(type ?? typeof(object), PostgresType, _serializerOptions);
             PgConverterInfo converterInfo;
             switch (DataFormat)
             {
-            case DataFormat.Binary:
-                // If we don't support binary we'll just throw.
+            case DataFormat.Text when IsUnknowResultType():
```

Comment:
Moved this code around a bit to make it clear this fallback mechanism only applies for UnknownResultTypes

Reply:
Hmm thinking about this a bit more, should we even want a 'fallback' mechanism here? We could just resolve directly for TextPgType if it's an unknown result type, essentially the user has told us they want reinterpretation, regardless of whether we may actually have a legitimate mapping as well. What do you think @roji ?